### PR TITLE
pc setup based on image vol2

### PR DESCRIPTION
**Pull request by ipa-fmw**
reopening https://github.com/ipa320/setup_cob4/pull/109 for further review

### CODE REVIEW

**Comment by ipa-nhg** [comment]
please see the comments: https://github.com/ipa320/setup_cob4/pull/109

File: images_config/ks-robot-master.cfg [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+echo "sources list:"
+cat /etc/apt/sources.list
+apt-get update
+apt-get upgrade -y
+sleep 20
+apt-get clean
+apt-get update
+aptitude install openssh-client=1:5.9p1-5ubuntu1.10
+apt-get install openssh-server -y
+sleep20
```
Comment (ipa-fmw) [inline images_config/ks-robot-master.cfg]: typo error : "sleep 20"

File: images_config/ks-robot-master.cfg [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+apt-get install aptitude -y
+apt-get update
+sleep 20
+apt-get clean
+apt-get update
+apt-get upgrade -y
+apt-get -f install 
+sleep20
```
Comment (ipa-fmw) [inline images_config/ks-robot-master.cfg]: typo error : "sleep 20"

File: images_config/ks-robot-slave.cfg [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+echo "sources list:"
+cat /etc/apt/sources.list
+apt-get update
+apt-get upgrade -y
+sleep 20
+apt-get clean
+apt-get update
+aptitude install openssh-client=1:5.9p1-5ubuntu1.10
+apt-get install openssh-server -y
+sleep20
```
Comment (ipa-fmw) [inline images_config/ks-robot-slave.cfg]: typo error : "sleep 20"

File: images_config/ks-robot-slave.cfg [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+apt-get install aptitude -y
+apt-get update
+sleep 20
+apt-get clean
+apt-get update
+apt-get upgrade -y
+apt-get -f install 
+sleep20
```
Comment (ipa-fmw) [inline images_config/ks-robot-slave.cfg]: typo error : "sleep 20"

File: images_config/ks-robot-master.cfg [hunk 5]
```diff
@@ -0,0 +1,370 @@
...
+
+###############################SETUP UDEV RULES############################################### 
+echo -e "\n${green}INFO: Setup udev rules${NC}\n"
+wget -O /etc/init.d/udev-cob.sh https://raw.githubusercontent.com/ipa-nhg-dd/setup_cob4/master/udev_rules/udev_cob.sh
```
Comment (ipa-fmw) [inline images_config/ks-robot-master.cfg]: use ipa320

File: images_config/ks-robot-master.cfg [hunk 6]
```diff
@@ -0,0 +1,370 @@
...
+echo -e "\n${green}INFO: Setup bash environment${NC}\n"
+sleep 5
+
+wget -O /etc/cob.bash.bashrc https://raw.githubusercontent.com/ipa320/setup_cob4/master/cob-pcs/cob.bash.bashrc.b
+echo -e "\n${green}INFO: Setup network interfaces${NC}\n"
+sleep 5
+
+
+ifdown eth0
+wget -O /etc/network/interfaces https://raw.githubusercontent.com/ipa-nhg-dd/setup_cob4/master/cob-pcs/networkInterfacesB1
```
Comment (ipa-fmw) [inline images_config/ks-robot-master.cfg]: use ipa320

File: images_config/ks-robot-master.cfg [hunk 7]
```diff
@@ -0,0 +1,370 @@
...
+############## GRAPHICS SETUP #############
+%post --interpreter /bin/bash
+echo "X11Forwarding yes" >> /etc/ssh/sshd_config
+echo "X11UseLocalhost no" >> /etc/ssh/sshd_config
+echo "PermitRootLogin yes">> /etc/ssh/sshd_config
+/etc/init.d/ssh restart
+%end
+
+############## NTP SETUP #############
+# TODO: use ntp templates from setup_cob4 repository
```
Comment (ipa-fmw) [inline images_config/ks-robot-master.cfg]: get template through
```
wget -O /etc/ntp.conf https://raw.githubusercontent.com/ipa320/setup_cob4/master/cob-pcs/ntp_server
sed -i 's/server_ip/$HOST/g' /etc/ssh/sshd_config
```

File: images_config/ks-robot-master.cfg [hunk 8]
```diff
@@ -0,0 +1,370 @@
...
+
+########### SETUP LOCAL ROOT USER #################
+%post
+echo "\n${green}INFO:Setup local ROOT user${NC}\n"
+echo "Defaults rootpw" >> /etc/sudoers
+rootpw robot
+#passwd root --iscrypted $1$tEHboVqc$8dkJ2OvCfg8P3z/ru2puI0
+echo "robot-local ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers
+echo "robot ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers
+apt-get -y -q update
```
Comment (ipa-fmw) [inline images_config/ks-robot-master.cfg]: not needed here, remove.

File: images_config/ks-robot-slave.cfg [hunk 9]
```diff
@@ -0,0 +1,356 @@
...
+echo $SERVERNAME
+touch /etc/auto.direct
+
+echo "/u  -fstype=nfs4    $SERVERNAME:/" >> /etc/auto.direct
+update-rc.d autofs defaults
+service autofs restart
+modprobe nfs
+
+echo "/-  /etc/auto.direct" >> /etc/auto.master
+ping -c 1 -w 3 $SERVERNAME
```
Comment (ipa-fmw) [inline images_config/ks-robot-slave.cfg]: this line make not sense, neither the if below.

please remove this block. we do not need to check if the server is available. we just configure it, that can be done offline

File: images_config/ks-robot-slave.cfg [hunk 10]
```diff
@@ -0,0 +1,356 @@
...
+############## GRAPHICS SETUP #############
+%post --interpreter /bin/bash
+echo "X11Forwarding yes" >> /etc/ssh/sshd_config
+echo "X11UseLocalhost no" >> /etc/ssh/sshd_config
+echo "PermitRootLogin yes">> /etc/ssh/sshd_config
+/etc/init.d/ssh restart
+%end
+
+############## NTP SETUP #############
+# TODO: use ntp templates from setup_cob4 repository
```
Comment (ipa-fmw) [inline images_config/ks-robot-slave.cfg]: same as for master
```
wget -O /etc/ntp.conf https://raw.githubusercontent.com/ipa320/setup_cob4/master/cob-pcs/ntp_client
sed -i 's/server_ip/$SERVERNAME/g' /etc/ssh/sshd_config
```

File: images_config/ks-robot-master.cfg (hunk 1 above)
Comment (ipa-nhg) [inline images_config/ks-robot-master.cfg]: already mentioned:  https://github.com/ipa320/setup_cob4/pull/109/files#r126953366

File: images_config/ks-robot-master.cfg (hunk 2 above)
Comment (ipa-nhg) [inline images_config/ks-robot-master.cfg]: already mentioned: https://github.com/ipa320/setup_cob4/pull/109/files#r126953350

File: images_config/ks-robot-master.cfg (hunk 5 above)
Comment (ipa-nhg) [inline images_config/ks-robot-master.cfg]: already mentioned: https://github.com/ipa320/setup_cob4/pull/109/files#r127136179

File: images_config/ks-robot-master.cfg (hunk 6 above)
Comment (ipa-nhg) [inline images_config/ks-robot-master.cfg]: already mentioned: https://github.com/ipa320/setup_cob4/pull/109/files#r127136218

File: images_config/ks-robot-master.cfg (hunk 8 above)
Comment (ipa-nhg) [inline images_config/ks-robot-master.cfg]: already mentioned: https://github.com/ipa320/setup_cob4/pull/109/files#r127137163

File: images_config/ks-robot-master.cfg (hunk 7 above)
Comment (ipa-nhg) [inline images_config/ks-robot-master.cfg]: already mentioned: https://github.com/ipa320/setup_cob4/pull/109/files#r127137613

File: images_config/ks-robot-slave.cfg (hunk 3 above)
Comment (ipa-nhg) [inline images_config/ks-robot-slave.cfg]: already mentioned:

File: images_config/ks-robot-slave.cfg (hunk 4 above)
Comment (ipa-nhg) [inline images_config/ks-robot-slave.cfg]: already mentioned: https://github.com/ipa320/setup_cob4/pull/109/files#r126953366

File: images_config/ks-robot-slave.cfg (hunk 9 above)
Comment (ipa-nhg) [inline images_config/ks-robot-slave.cfg]: already mentioned: https://github.com/ipa320/setup_cob4/pull/109/files#r127137457

File: images_config/ks-robot-slave.cfg (hunk 10 above)
Comment (ipa-nhg) [inline images_config/ks-robot-slave.cfg]: already mentioned: https://github.com/ipa320/setup_cob4/pull/109/files#r127137613

**Comment by ipa-nhg** [comment]
@ipa-fmw please use the old PR to add comments https://github.com/ipa320/setup_cob4/pull/109/files#r127137613 , and don't copy here again my review

File: images_config/README_images.md [hunk 11]
```diff
@@ -327,69 +326,53 @@ d-i partman-md/confirm_nooverwrite      boolean true
...
-mkdir ubuntu_iso_slave
-sudo  mount -o loop ~/Downloads/ubuntu-14.04.5-server-amd64.iso ubuntu_iso_master
-sudo  mount -o loop ~/Downloads/ubuntu-14.04.5-server-amd64.iso ubuntu_iso_slave
-```
-Copy .iso contents to another folder on your desktop so we can edit the files. Don't forget to set the right permissions to be able to make changes.
+mkdir ~/ubuntu_iso_master
+sudo  mount -o loop ~/Downloads/ubuntu-14.04.5-server-amd64.iso ~/ubuntu_iso_master
+cp -r ~/ubuntu_iso_master ~/ubuntu_files_master
+chmod +w -R ~/ubuntu_files_master
+cp ~/git/setup_cob4/images_config/ks-robot-master.cfg ~/ubuntu_files_master/ks-robot.cfg
```
Comment (ipa-nhg) [inline images_config/README_images.md]: previously:
`
 git clone https://github.com/ipa320/setup_cob4  ~/git/`

File: images_config/ks-robot-master.cfg [hunk 12]
```diff
@@ -123,72 +121,44 @@ glib-compile-schemas $settingsDir
...
-#apt-get -y autoremove
-#apt-get clean
-
 # Add your custom post installation script here. It's not necessary, but you can add something depending on your needs. Here's my script:
 ######################POST-INSTALLATION#######################################
+
+########## SETUP GRUB ##################
 %post --interpreter=/bin/bash
 #!/bin/bash
-apt-get update && apt-get upgrade -y
```
Comment (ipa-nhg) [inline images_config/ks-robot-master.cfg]: do not remove this line, you should upgrade the packages before you start with the configuration.

**Comment by ipa-fmw** [comment]
I'l close this because changes are merged in #120

**Pull request closed without merging**